certbot/storage.py

```diff
@@ -117,10 +125,24 @@ def write_renewal_config(o_filename, n_filename, archive_dir, target, relevant_d
     # TODO: add human-readable comments explaining other available
     #       parameters
     logger.debug("Writing new config %s.", n_filename)
+
+    # Ensure that the file exists
+    open(n_filename, 'a').close()
+
+    # Copy permissions from the old version of the file. If writing a new file, or if the file is
+    # old, ensure that the file can only be accessed by the user. (Equivalent to `chmod go-rwx`.)
+    current_permissions = stat.S_IMODE(os.lstat(o_filename).st_mode)
+    if restrict_permissions:
+        new_permissions = current_permissions & ~stat.S_IRWXG & ~stat.S_IRWXO
+    else:
+        new_permissions = current_permissions
+    os.chmod(n_filename, new_permissions)
```

Comment:
During each renewal, a new renewal conf file is created (and, later, renamed to have the same name as the existing file, so that updates are atomic).

Initially, the new file's permissions seem to be based on the `umask`.

The `else` case ensures that the old file's permissions are preserved (from the user's point of view) even though a new file is created as an implementation detail.